### PR TITLE
addreses #8323 : --runnableExamplesDir:mydir to keep git status clean

### PR DESCRIPTION
**Pull request by timotheecour**
* addreses issue #8323 : --runnableExamplesDir:mydir to keep git status clean

### CODE REVIEW

**Comment by Varriount** [comment]
Do runnableExamples normally generate output files?

**Comment by timotheecour** [comment]
yes; and then they get deleted on success; but stay on failure. 
* this avoids both over-writing some potentially already existing file named foo_examples.nim (however unlikely) 
* more importantly, allows `git status` to stay clean even on failure, when user passes `--runnableExamplesDir:mydir`

File: compiler/sem.nim [hunk 1]
```diff
@@ -609,15 +609,19 @@ proc myProcess(context: PPassContext, n: PNode): PNode =
...
   let inp = toFullPath(c.config, c.module.info)
-  let outp = inp.changeFileExt"" & "_examples.nim"
+  let runnableExamplesDir = c.config.runnableExamplesDir
+  var outp = inp.changeFileExt"" & "_examples.nim"
+  if runnableExamplesDir.len>0:
+    discard existsOrCreateDir(runnableExamplesDir)
+    outp = runnableExamplesDir / outp.extractFilename
   let nimcache = outp.changeFileExt"" & "_nimcache"
-  renderModule(c.runnableExamples, inp, outp)
+  renderModule(c.runnableExamples, inp, outp, conf = c.config)
```
Comment (timotheecour) [inline compiler/sem.nim]: NOTE to reviewer: Please double check this line where I added `conf = c.config`
was the intention to have nil conf?

Comment (Araq) [reply compiler/sem.nim]: Your change seems to be correct.

Comment (timotheecour) [reply compiler/sem.nim]: ok to merge then?

**Comment by Araq** [comment]
Why do we need this? If your runnable example fail, a dirty ``git status`` should be the least of your worries. And even if we need this, why not use the ``nimcache`` dir for it? And also, a pattern like ``*_example.nim`` in your ``.gitignore`` can also handle it.

**Comment by GULPF** [comment]
IMO they should use the `nimcache` dir. #8281 could work by creating subfolders within `nimcache`?

**Comment by dom96** [comment]
> why not use the nimcache dir for it?

> IMO they should use the nimcache dir

I also think they should use the nimcache dir.

**Comment by timotheecour** [comment]
so nested nimcaches? eg:
`nimcache/[foo_nimcache,foo_examples.nim, ,foo_examples]`
?

**Comment by Araq** [comment]
> it becomes my worry once the runnableExamples failure is fixed

how so? the examples will be removed then. But anyway, how ``nimcache`` works will be changed.

**Comment by timotheecour** [comment]
> the examples will be removed then

indeed, I take back that particular point; I still want git status to remain clean even on `nim doc foo` failure though.

I'm closing this PR and opened a new PR that implements the suggestion of rooting generated files (`foo_nimcache,foo_examples.nim, ,foo_examples`) under `nimcache` hierarchy; see https://github.com/nim-lang/Nim/pull/8415

**Pull request closed without merging**